Lab1_Operationalizing_Pytorch_with_Mlflow/src/infra/prov/nodes.tf

```
@@ -0,0 +1,108 @@
+resource "google_compute_instance" "training_node" {
+  name                      = "training-node"
+  machine_type              = "c2-standard-8"
+  tags                      = ["ssh", "web", "tf-created"]
+  zone                      = var.gcp_zone
+  allow_stopping_for_update = true
+
+  boot_disk {
+    initialize_params {
+      image = "ubuntu-2004-focal-v20210105"
+      size  = 100
+      type  = "pd-ssd"
+    }
+  }
+
+  network_interface {
+    subnetwork = google_compute_subnetwork.englab_subnet.name
+
+    access_config {
+      nat_ip = google_compute_address.static_ip["training"].address
+    }
+  }
+
+  metadata = {
+    "ssh-keys" = "${var.ansible_ssh.user}:${file(var.ansible_ssh.key_file)}"
+  }
+
+  /* So now we can seamless use registry and storage APIs from this instance
+   * Check iam.tf to loock for its definition
+   */
+  service_account {
+    email  = google_service_account.mlflow_data_account.email
+    scopes = ["cloud-platform"]
+  }
+}
+
+resource "google_compute_instance" "tracking_node" {
+  name                      = "tracking-node"
+  machine_type              = "g1-small"
+  tags                      = ["ssh", "tracking", "tf-created"]
+  zone                      = "southamerica-east1-a"
+  allow_stopping_for_update = true
+
+  boot_disk {
+    initialize_params {
+      image = "ubuntu-2004-focal-v20210105"
+      size  = 100
+      type  = "pd-ssd"
```

tracker doesn't store any data so can use smaller, cheaper storage

---

Base image has 10GB.  What about 15GB? So, if we somehow we need extra space during conf and installation of something we "won't be catch with a short blanket".

```suggestion
      size  = 15
      type  = "pd-ssd"
```